Dockerfile.base

```diff
@@ -0,0 +1,33 @@
+FROM registry.access.redhat.com/ubi9/go-toolset:1.20 as builder
+
+USER 0
+
+RUN yum -y install yum-utils
+RUN yum-config-manager --enable ubi-9-baseos-source
+
+WORKDIR /elfutils-source
+RUN yumdownloader --source elfutils
+RUN yum -y install cpio
+RUN rpm2cpio elfutils-0.189-3.el9.src.rpm | cpio -iv
+RUN tar xjvf elfutils-0.189.tar.bz2
+WORKDIR /elfutils-source/elfutils-0.189
+RUN ./configure --disable-debuginfod
+RUN make install
+
+
+WORKDIR /libbpf-source
+RUN yumdownloader --source libbpf
+RUN rpm2cpio libbpf-1.2.0-1.el9.src.rpm | cpio -iv
+RUN tar xf ./linux-*el9.tar.xz
+WORKDIR /libbpf-source/linux-5.14.0-333.el9/tools/lib/bpf
+RUN make install_headers
+RUN prefix=/usr BUILD_STATIC_ONLY=y make install
+WORKDIR /libbpf-source/linux-5.14.0-333.el9/tools/bpf
+RUN make bpftool
+
+RUN yum -y install clang
```

Comment:
can you move build tools to the builder image? the base image only has the runtime dependency for kepler.

Reply:
@rootfs the `Dockerfile.base` is the builder image. may be it should be renamed as such?

Reply:
we have 3 dockerfiles: base, builder, and kepler

The build tools are in the builder image, e.g. this one https://github.com/sustainable-computing-io/kepler/blob/main/build/Dockerfile.libbpf.builder

Reply:
but in my refactoring there are only two Dockerfiles. `Dockerfile.base` which has all the things needed to build kepler. and `Dockerfile` which has a stage to build kapler and next stage copies everything needed to run kepler.
https://github.com/sustainable-computing-io/kepler/pull/1068

Reply:
Got it, so the new base is just ubi9 mimal.